### PR TITLE
Address more stacktrace print issues

### PR DESCRIPTION
**Pull request by sywhang**
Follow-up PR to address more issues for #14867. Fixes newline issues for ```Debug.Assert``` and broken error message/stacktrace for Contract failures.

### CODE REVIEW

**Comment by sywhang** [comment]
With this fix, the following program prints this:
```
Assertion Failed
some message

   at test.Program.Main(String[] args)
Aborted (core dumped)
```
Program.cs:
```
    class Program
    {
        static void Main(string[] args)
        {
            Debug.Assert(false, "some message");
        }
    }

```

And for the excess newline issue on ```Debug.Assert(false)``` is also addressed:
```
Assertion Failed

   at test.Program.Main(String[] args)
```
Program.cs:
```
    class Program
    {
        static void Main(string[] args)
        {
            Debug.Assert(false);
        }
    }
```

Additionally, ```Contract.Assume``` was printing strange things: 
```
```

**Comment by sywhang** [comment]
cc @jkotas @danmosemsft

File: src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs [hunk 1]
```diff
@@ -21,7 +21,19 @@ private static void ShowDialog(string stackTrace, string message, string detailM
                 // In Core, we do not show a dialog.
                 // Fail in order to avoid anyone catching an exception and masking
                 // an assert failure.
-                var ex = new DebugAssertException(message, detailMessage, stackTrace);
+                DebugAssertException ex;
+                if (message == String.Empty) 
+                {
+                    ex = new DebugAssertException(stackTrace);
```
Comment (stephentoub) [inline src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs]: What if detailMessage contained meaningful information? Doesn't this lose that?

File: src/mscorlib/src/System/Diagnostics/Debug.Windows.cs [hunk 2]
```diff
@@ -17,7 +17,19 @@ private static void ShowDialog(string stackTrace, string message, string detailM
                 // In Core, we do not show a dialog.
                 // Fail in order to avoid anyone catching an exception and masking
                 // an assert failure.
-                var ex = new DebugAssertException(message, detailMessage, stackTrace);
+                DebugAssertException ex;
+                if (message == String.Empty) 
+                {
+                    ex = new DebugAssertException(stackTrace);
```
Comment (stephentoub) [inline src/mscorlib/src/System/Diagnostics/Debug.Windows.cs]: Same question

File: src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs (hunk 1 above)
Comment (sywhang) [reply src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs]: This gets called from the 3 method overrides of Debug.Assert():
public static void Assert(bool condition)
public static void Assert(bool condition, string message)  
public static void Assert(bool condition, string message, string detailMessage)
and we just pass string.Empty for missing parameters... 
I suppose we would lose detailMessage if a user does something like:
```
Debug.Assert(false, String.Empty, "some detailed message");
```
While I see such case as somewhat unlikely, I do see the need to fix that for completeness so thanks for pointing that out :)

File: src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs [hunk 3]
```diff
@@ -21,7 +21,19 @@ private static void ShowDialog(string stackTrace, string message, string detailM
                 // In Core, we do not show a dialog.
                 // Fail in order to avoid anyone catching an exception and masking
                 // an assert failure.
-                var ex = new DebugAssertException(message, detailMessage, stackTrace);
+                DebugAssertException ex;
+                if (message == String.Empty)
```
Comment (stephentoub) [inline src/mscorlib/shared/System/Diagnostics/Debug.Unix.cs]: Also, what if the user passes in null for these? Should there be string.IsNullOrEmpty? IsNullOrWhiteSpace?

**Pull request merged**